k8s/operators/src/diskpool/main.rs

```diff
@@ -619,20 +645,19 @@ impl ResourceContext {
 /// is wrong would be to consult the logs.
 async fn ensure_crd(k8s: Client) {
     let dsp: Api<k8s_openapi::apiextensions_apiserver::pkg::apis::apiextensions::v1::CustomResourceDefinition> = Api::all(k8s);
-    let lp = ListParams::default().fields(&format!("metadata.name={}", "diskpools.openebs.io"));
-    let crds = dsp.list(&lp).await.expect("failed to list CRDS");
 
-    // the CRD has not been installed yet, to avoid overwriting (and create upgrade issues) only
-    // install it when there is no crd with the given name
-    if crds.iter().count() == 0 {
-        let crd = DiskPool::crd();
+    // Check if there is an existing dsp crd. If yes, Replace it with new generated one.
+    // Create new if it doesnt exist.
+    let mut crd = DiskPool::crd();
+    if let Ok(exiting) = dsp.get("diskpools.openebs.io").await {
```

Comment:
I know it was already hardcoded like so, but can we retrieve this value from somewhere instead, eg: `CRD` type?

Reply:
now getting crd name from the crd instance that gets created on line 651